lib/sidekiq-bus.rb

```diff
@@ -1,6 +1,38 @@
-require "queue-bus"
-require "sidekiq_bus/adapter"
-require "sidekiq_bus/version"
+# frozen_string_literal: true
+
+require 'queue-bus'
+require 'sidekiq_bus/adapter'
+require 'sidekiq_bus/version'
 require 'sidekiq_bus/middleware/retry'
 
 QueueBus.adapter = QueueBus::Adapters::Sidekiq.new
+
+module SidekiqBus
+  # This method will analyze the current queues and generate an array that
+  # can operate as the sidekiq queues configuration. It should be based on how
+  # The sidekiq CLI builds weighted queues.
+  #
+  # @param overrides [Hash<String, Integer>] A mapping of queue names and
+  #     weights that must be included
+  # @param default [Integer] The default weight to apply to any given queue
+  # @returns [Array<String>] The set of queue names weighted to sidekiq
+  def self.generate_weighted_queues(overrides: {}, default: 1)
+    queues = Set.new(QueueBus::TaskManager.new(false).queue_names)
+
+    # Also pitches-in for driving the bus.
+    queues << 'bus_incoming'
+    queues += overrides.keys
+
+    # Sorts by weight to provide a visual indication of queue order in sidekiq
+    # UI. Otherwise they can appear in various orders.
+    queues.sort_by { |queue| overrides[queue] || default }
+
+    queues.flat_map do |queue|
+      # Constrains the weight to 1 or greater
+      weight = [1, overrides[queue] || default].max
```

Comment:
should this line match the sorting thing just above?

Reply:
maybe you want to sort by this, then alphabetical just so it really is always the same.